### PR TITLE
refactor: gallery component + usage

### PR DESCRIPTION
**Pull request by drewpledger**
Refactored the Gallery component and its usages.

https://github.com/user-attachments/assets/64684340-ad1e-4747-89f7-87db44396f26

### CODE REVIEW

File: core/components/ui/gallery/gallery.tsx [hunk 1]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
   images: Image[] | [];
   defaultImageIndex?: number;
+  noImageText?: string;
+  asChild?: boolean;
 }
 
-const Gallery = forwardRef<ElementRef<'div'>, GalleryProps>(
-  ({ className, children, images, defaultImageIndex = 0, ...props }, ref) => {
+export const Gallery = forwardRef<ElementRef<'div'>, GalleryProps>(
+  ({ className, children, images, asChild, defaultImageIndex = 0, noImageText, ...props }, ref) => {
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:74]: Remove `asChild` as it is not used

File: core/components/ui/gallery/gallery.tsx [hunk 2]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+
     return (
-      <GalleryContext.Provider value={{ images, selectedImageIndex, setSelectedImageIndex }}>
-        <div aria-live="polite" className={cn(className)} ref={ref} {...props}>
-          {children}
-        </div>
-      </GalleryContext.Provider>
+      <div aria-live="polite" className={cn(className)} ref={ref} {...props}>
+        <figure
+          className={cn('relative aspect-square h-full max-h-[548px] w-full', className)}
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:86]: remove `className`

File: core/components/ui/gallery/gallery.tsx [hunk 3]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+          ref={ref}
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:87]: remove `ref`

File: core/components/ui/gallery/gallery.tsx [hunk 4]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+          {...props}
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:88]: remove prop spread

Comment (jorgemoya) [inline core/components/ui/gallery/gallery.tsx:88]: Remove ref and props.

File: core/components/ui/gallery/gallery.tsx [hunk 5]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+          ) : (
+            <div className="flex aspect-square items-center justify-center bg-gray-200">
+              <div className="text-base font-semibold text-gray-500">
+                {noImageText ?? 'Coming soon'}
+              </div>
+            </div>
+          )}
+          {images.length > 1 && (
+            <div
+              className={cn(
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:108]: remove `className`

File: core/components/ui/gallery/gallery.tsx [hunk 6]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+                'absolute top-1/2 flex w-full -translate-y-1/2 justify-between px-5 sm:px-0',
+                className,
+              )}
+              ref={ref}
+              {...props}
+            >
+              {children || (
```
Comment (jorgemoya) [inline core/components/ui/gallery/gallery.tsx:115]: Remove children and just render indicators.

File: core/components/ui/gallery/gallery.tsx [hunk 7]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+              </div>
+            </div>
+          )}
+          {images.length > 1 && (
+            <div
+              className={cn(
+                'absolute top-1/2 flex w-full -translate-y-1/2 justify-between px-5 sm:px-0',
+                className,
+              )}
+              ref={ref}
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:112]: remove `ref`

File: core/components/ui/gallery/gallery.tsx [hunk 8]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+              {...props}
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:113]: remove prop spread

File: core/components/ui/gallery/gallery.tsx [hunk 9]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+                className={cn(
+                  'inline-block h-12 w-12 flex-shrink-0 flex-grow-0 focus-visible:outline-none focus-visible:ring-4 focus-visible:ring-primary/20 md:h-24 md:w-24',
+                  className,
+                )}
+                onClick={(e) => {
+                  setSelectedImageIndex(index);
+                }}
+                type="button"
+              >
+                <Slot
```
Comment (jorgemoya) [inline core/components/ui/gallery/gallery.tsx:163]: Use div

File: core/components/ui/gallery/gallery.tsx [hunk 10]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+                    }
+                  />
+                </>
+              )}
+            </div>
+          )}
+        </figure>
+        <nav
+          aria-label="Thumbnail navigation"
+          className={cn(
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:144]: remove `className`

File: core/components/ui/gallery/gallery.tsx [hunk 11]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+            'mt-3 flex w-full flex-wrap items-center gap-4 px-1 py-1 md:mt-5 md:gap-6',
+            className,
+          )}
+        >
+          {images.map((image, index) => {
+            return (
+              <button
+                aria-label="Enlarge product image"
+                aria-pressed={selectedImageIndex === index}
+                className={cn(
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:154]: remove `className`

File: core/components/ui/gallery/gallery.tsx [hunk 12]
```diff
@@ -90,209 +63,124 @@ const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWith
...
+                  'inline-block h-12 w-12 flex-shrink-0 flex-grow-0 focus-visible:outline-none focus-visible:ring-4 focus-visible:ring-primary/20 md:h-24 md:w-24',
+                  className,
+                )}
+                onClick={(e) => {
+                  setSelectedImageIndex(index);
+                }}
+                type="button"
+              >
+                <Slot
+                  className={cn(
```
Comment (matthewvolk) [inline core/components/ui/gallery/gallery.tsx:164]: remove `cn` and refactor to get rid of `Slot`

**Pull request closed without merging**